hack/testing-olm-upgrade/test-020-olm-upgrade.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# Given an OLM manifest, verify a green field deployment
+# of cluster logging by asserting CLO creates the resources
+# that begets the operands that make up logging.
+
+set -euo pipefail
+
+repo_dir="$( cd "$(dirname "$0")/../.." ; pwd -P )"
+source "$repo_dir/hack/testing-olm/utils"
+source "$repo_dir/hack/testing-olm/assertions"
+
+start_seconds=$(date +%s)
+
+ARTIFACT_DIR=${ARTIFACT_DIR:-"$repo_dir/_output/"}
+ARTIFACT_DIR="${ARTIFACT_DIR}/$(basename ${BASH_SOURCE[0]})"
+if [ ! -d $ARTIFACT_DIR ] ; then
+  mkdir -p $ARTIFACT_DIR
+fi
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+TIMEOUT_MIN=$((2 * $minute))
+export CLUSTER_LOGGING_OPERATOR_NAMESPACE="openshift-logging"
+manifest_dir=${repo_dir}/manifests
+version=$(basename $(find $manifest_dir -type d | sort -r | head -n 1))
+#use N-2 since OLM content is not matched to OCP version until release
+previous_version=$(echo $version | awk '{print $1 - 0.1}')
+
+cleanup(){
+  local return_code="$?"
+  set +e
+  log::info "Running cleanup"
+  end_seconds=$(date +%s)
+  runtime="$(($end_seconds - $start_seconds))s"
+  oc -n openshift-operators-redhat -o yaml get subscription elasticsearch-operator > $ARTIFACT_DIR/subscription-eo.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator-deployment.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get pods > $ARTIFACT_DIR/openshift-operators-redhat-pods.yml 2>&1 ||:
+  oc -n openshift-operators-redhat -o yaml get configmaps > $ARTIFACT_DIR/openshift-operators-redhat-configmaps.yml 2>&1 ||:
+  oc -n openshift-operators-redhat describe deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator.describe 2>&1 ||:
+
+  oc logs -n "openshift-operators-redhat" deployment/elasticsearch-operator > $ARTIFACT_DIR/elasticsearch-operator.log 2>&1 ||:
+  oc  -n openshift-operator-lifecycle-manager logs --since=$runtime deployment/catalog-operator > $ARTIFACT_DIR/catalog-operator.logs 2>&1 ||:
+  oc  -n openshift-operator-lifecycle-manager logs --since=$runtime deployment/olm-operator > $ARTIFACT_DIR/olm-operator.logs 2>&1 ||:
+
+  ${repo_dir}/olm_deploy/scripts/operator-uninstall.sh
+  ${repo_dir}/olm_deploy/scripts/catalog-uninstall.sh
+
+  set -e
+  exit ${return_code}
+}
+trap cleanup exit
+
+
+# deploy elasticsearch-operator
+log::info "Deploying elasticsearch-operator ${previous_version} from marketplace..."
```

Comment:
if, and it may be big if, the 4.4 and 4.5 CSVs are identical, you could go ahead and use the 4.5 "deploy a new catalog source" approach to deploy a catalog source that services up a 4.5 CSV, but with the 4.4 images substituted into it.

It's a bit icky since it isn't using the 4.4 CSV, but it puts you on a better path towards the "right" way to deploy the N-1 ES bits (rather than having to wait for them to be in the public marketplace to be able to install them)